### PR TITLE
Translate

### PR DESCRIPTION
**Pull request by AbbasKDG**
translation module

### CODE REVIEW

**Comment by Harukine** [comment]
I'm sorry, how do we use this??

**Review by Harukine — CHANGES_REQUESTED** [review]
We need a function that can be called

**Review by aeciorc — CHANGES_REQUESTED** [review]
Like @Harukine mentioned, you will need to refactor this as a function. It should take the text, target language, source language (optional, auto-detect if it's null) and return the result.

File: server/.gitignore [hunk 1]
```diff
@@ -167,4 +167,5 @@ $RECYCLE.BIN/
 # Windows shortcuts
 *.lnk
 
-# End of https://www.gitignore.io/api/node,linux,macos,windows,visualstudiocode
\ No newline at end of file
+# End of https://www.gitignore.io/api/node,linux,macos,windows,visualstudiocode
+.env
```
Comment (aeciorc) [inline server/.gitignore]: You can remove this, it was already on line 110

File: server/lib/testTranslate.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const translator = require("./translator");
+
+const text = "Hey";
+// The target language, we choose Russian as an example
+// language documentation: https://cloud.google.com/translate/docs/languages
+const target = "ru";
+
+const translation = translator(target, text);
+
```
Comment (aeciorc) [inline server/lib/testTranslate.js]: Remove this file from the repo, since it's not used

File: server/lib/translator.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// Allows to access env variables
+const dotenv = require("dotenv").config();
```
Comment (aeciorc) [inline server/lib/translator.js]: You don't want this here

File: server/lib/translator.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+// // Add in local .env file
+// // Google project_id for google translate api
+//projectId=""
+//// Imports the Google Cloud client library
+//key=""
+
+async function quickStart(target, text) {
```
Comment (aeciorc) [inline server/lib/translator.js]: change the name of this function

File: server/lib/translator.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  const projectId = process.env.projectId;
+
+  const key = process.env.key;
+  const { Translate } = require("@google-cloud/translate");
```
Comment (aeciorc) [inline server/lib/translator.js]: convert this to an import statement and put it on top of the file

File: server/lib/translator.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+  // Instantiates a client
+  const translate = new Translate({
```
Comment (aeciorc) [inline server/lib/translator.js]: You can instantiate this outside of the function

File: server/lib/translator.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    projectId,
+    key
+  });
+
+  const [translation] = await translate.translate(text, target);
```
Comment (aeciorc) [inline server/lib/translator.js]: You want to wrap this with a try/catch block

**Review by Disquietude — CHANGES_REQUESTED** [review]
Why are you editing the front-end components? The translate function exists only in the back-end, and I've already finished those components.

**Comment by Harukine** [comment]
Do not merge this PR

File: .gitignore [hunk 8]
```diff
@@ -167,4 +167,7 @@ $RECYCLE.BIN/
 # Windows shortcuts
 *.lnk
 
-# End of https://www.gitignore.io/api/node,linux,macos,windows,visualstudiocode
\ No newline at end of file
+# End of https://www.gitignore.io/api/node,linux,macos,windows,visualstudiocode
+
+.env
```
Comment (aeciorc) [inline .gitignore:172]: .env was already there

File: package.json [hunk 9]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (aeciorc) [inline package.json]: Delete this file

File: package-lock.json [hunk 10]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (aeciorc) [inline package-lock.json]: Delete this file

File: server/.gitignore (hunk 1 above)
Comment (aeciorc) [reply server/.gitignore]: It's still there

File: server/lib/package-lock.json [hunk 11]
```diff
@@ -0,0 +1,609 @@
+{
```
Comment (aeciorc) [inline server/lib/package-lock.json]: delete  this file

File: server/lib/package.json [hunk 12]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (aeciorc) [inline server/lib/package.json]: delete this file

File: server/lib/translator.js (hunk 4 above)
Comment (aeciorc) [reply server/lib/translator.js]: Still need to change the name

File: server/lib/translator.js [hunk 13]
```diff
@@ -0,0 +1,33 @@
+// Allows to access env variables
```
Comment (aeciorc) [inline server/lib/translator.js]: Remove these comments

File: server/models/Contact.js [hunk 14]
```diff
@@ -1,31 +1,34 @@
...
+      required: true
     }
-}, {timestamps: true});
+  },
+  { timestamps: true }
+);
 
 // Export the model so we can access it outside of this file
-module.exports = Contact = mongoose.model('contacts', ContactSchema);
+module.exports = Contact = mongoose.model("contacts", ContactSchema);
```
Comment (aeciorc) [inline server/models/Contact.js:34]: The team needs to use the same code formatter to avoid diffs like this

**Review by aeciorc — CHANGES_REQUESTED** [review]
I made some comments. You still need to address my comments from the last review

File: server/lib/translator.js (hunk 7 above)
Comment (AbbasKDG) [reply server/lib/translator.js]: Done

File: server/lib/translator.js (hunk 6 above)
Comment (AbbasKDG) [reply server/lib/translator.js]: Done

File: server/lib/translator.js (hunk 5 above)
Comment (AbbasKDG) [reply server/lib/translator.js]: According to my observation import statements are not my team's practice on the server side.

Comment (aeciorc) [reply server/lib/translator.js]: I see, that's fair

File: server/lib/translator.js [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+  projectId,
+  key
+});
+
+async function translateService(target, text) {
+  // Instantiates a client
+  try {
+    const [translation] = await translate.translate(text, target);
+    //Print text
+    console.log(`Text: ${text}`);
```
Comment (aeciorc) [inline server/lib/translator.js]: Remove these log

File: server/lib/translator.js [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+    //Print translation
+    console.log(`Translation: ${translation}`);
+    return translation;
+  } catch (err) {
+    alert(err);
```
Comment (aeciorc) [inline server/lib/translator.js]: Here you can log, remove the alert

**Review by Disquietude — CHANGES_REQUESTED** [review]
The package.json files in the server folder are important. The unnecessary package.json files in the root directory are the ones that should be deleted.

**Comment by Harukine** [comment]
Please do not merge, no other files other than translator.js should have been modified.

**Review by Harukine — CHANGES_REQUESTED** [review]
Too many unnecessary breaking changes, only translator.js needed to be worked on. Server cannot be ran on this branch.

**Pull request closed without merging**